src/pkg/services/m365/api/channels.go

```diff
@@ -1 +1,376 @@
 package api
+
+import (
+	"context"
+
+	"github.com/alcionai/clues"
+	"github.com/alcionai/corso/src/internal/common/ptr"
+	"github.com/alcionai/corso/src/internal/m365/graph"
+	"github.com/alcionai/corso/src/pkg/backup/details"
+	"github.com/alcionai/corso/src/pkg/fault"
+	"github.com/alcionai/corso/src/pkg/logger"
+	"github.com/microsoft/kiota-abstractions-go/serialization"
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+	"github.com/microsoftgraph/msgraph-sdk-go/teams"
+)
+
+// ---------------------------------------------------------------------------
+// controller
+// ---------------------------------------------------------------------------
+
+func (c Client) Channels() Channels {
+	return Channels{c}
+}
+
+// Channels is an interface-compliant provider of the client.
+type Channels struct {
+	Client
+}
+
+// ---------------------------------------------------------------------------
+// containers
+// ---------------------------------------------------------------------------
+
+// CreateContainer makes an channels with the name in the team
```

Comment:
Is this something we need right away?  Since we're going to start off with Export instead of Restore for channels, I think we can skip channel creation.